PAM_app/app/src/main/java/com/example/pam_app/cosmetics/CosmeticsActivity.kt

```diff
@@ -16,68 +16,64 @@ import com.example.pam_app.cosmetics.presenter.Presenter
+        presenter = Presenter(this, Model(Repository()))
+        setupRecyclerview()
+        presenter.initView()
+    }
 
-        override fun init() {
-            System.out.println("beggin post request")
-            val myPost = Cosmetics(
-                "https://cdn.shopify.com/s/files/1/1338/0845/products/brain-freeze_a_800x1200.jpg?v=1502255076",
-                "Lipstick", 10.2F
-            )
-            presenter.putItem(myPost)
+    override fun init() {
+        System.out.println("beggin post request")
+        val myPost = Cosmetics(
+            "https://cdn.shopify.com/s/files/1/1338/0845/products/brain-freeze_a_800x1200.jpg?v=1502255076",
+            "Lipstick", 10.2F
+        )
+        presenter.putItem(myPost)
 
-            val myPost2 = Cosmetics(
-                "https://www.purpicks.com/wp-content/uploads/2018/02/Ombre-Amazonie-CC.png",
-                "Eyeshadow", 5.2F
-            )
-            presenter.postSmth(myPost2)
-            presenter.getBlush()
-            presenter.getLipstick()
-        }
+        val myPost2 = Cosmetics(
+            "https://www.purpicks.com/wp-content/uploads/2018/02/Ombre-Amazonie-CC.png",
+            "Eyeshadow", 5.2F
+        )
+        presenter.postSmth(myPost2)
+        presenter.getBlush()
+        presenter.getLipstick()
+    }
 
-        override fun showGetResponse(response: Response<List<Cosmetics>>) {
-            if (response.isSuccessful) {
-                response.body()?.let { adapter.setData(it) }
-            } else {
-                Toast.makeText(this, response.code().toString(), Toast.LENGTH_SHORT).show()
-            }
+    override fun showGetResponse(response: Response<List<Cosmetics>>) {
+        if (response.isSuccessful) {
+            response.body()?.let { adapter.setData(it) }
+        } else {
+            Toast.makeText(this, response.code().toString(), Toast.LENGTH_SHORT).show()
         }
+    }
 
-        override fun showPostResponse(response: Response<Cosmetics>) {
-            if (response.isSuccessful) {
-                println("successful")
-                Log.d("Main", response.body().toString())
-                Log.d("Main", response.code().toString())
-                Log.d("Main", response.message())
-            } else {
-                println("unsuccessful POST request - doesn't have the access to the api")
-                Toast.makeText(this, response.code().toString(), Toast.LENGTH_SHORT).show()
-            }
+    override fun showPostResponse(response: Response<Cosmetics>) {
+        if (response.isSuccessful) {
+            println("successful")
+            Log.d("Main", response.body().toString())
+            Log.d("Main", response.code().toString())
+            Log.d("Main", response.message())
+        } else {
+            println("unsuccessful POST request - doesn't have the access to the api")
+            Toast.makeText(this, response.code().toString(), Toast.LENGTH_SHORT).show()
         }
+    }
```

Comment:
Same problem with presentation business logic here. Also don't use `Response` data type in Activities, use only plain models.